src/topp/DatabaseSuitability.cpp

```diff
@@ -0,0 +1,367 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Tom Waschischeck $
+// $Authors: Tom Waschischeck $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/APPLICATIONS/TOPPBase.h>
+
+#include <OpenMS/CONCEPT/LogStream.h>
+#include <OpenMS/FORMAT/IdXMLFile.h>
+#include <OpenMS/FORMAT/MzMLFile.h>
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <algorithm>
+#include <cmath>
+#include <cstdio>
+
+using namespace OpenMS;
+using namespace std;
+
+//-------------------------------------------------------------
+// Doxygen docu
+//-------------------------------------------------------------
+
+/**
+@page TOPP_DatabaseSuitability DatabaseSuitability
+
+@brief Calculates a suitability for a database which was used a for peptide identification search. Also reports the quality of LC-MS spectra.
+
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+class DatabaseSuitability :
+  public TOPPBase
+{
+public:
+  DatabaseSuitability() :
+    TOPPBase("DatabaseSuitability", "Computes a suitability score for a database which was used for a peptide identification search. Also reports the quality of LC-MS spectra.", false)
+  {
+  }
+
+protected:
+
+  // this function will be used to register the tool parameters
+  // it gets automatically called on tool execution
+  void registerOptionsAndFlags_() override
+  {
+    registerInputFile_("in_id", "<file>", "", "Input idXML file from peptide search (after FDR)");
+    setValidFormats_("in_id", { "idXML" });
+    registerInputFile_("in_spec", "<file>", "", "Input MzML file");
+    setValidFormats_("in_spec", { "mzML" });
+    registerInputFile_("in_novo", "<file>", "", "Input idXML file containing de novo peptides");
+    setValidFormats_("in_novo", { "idXML" });
+    registerOutputFile_("out", "<file>", "", "Optional tsv output", false); //tsv output
+    setValidFormats_("out", { "tsv" });
+    registerIntOption_("novor_fract", "<integer>", 1, "Set the percentage of de novo peptides to capture with a score higher than the fasta score.", false, true);
+    registerFlag_("force_no_re_rank", "Use this flag if you want to disable re-ranking. This might yeild in underperformance.", true);
```

Comment:
what is re-ranking? Short description here and longer description in the docs.
typo in `yeild`, but `underestimate quality` is probably more accurate?